### PR TITLE
PAM improvements

### PR DESCRIPTION
**Pull request by alepintus**
Hi @joewhite101 @eugeniosant ,
this is the cherry-pick from [this PR](https://github.com/entando/entando-components/pull/394).
Could you please have a look?
Thanks!

List of commits imported:
* Improvement of instance selector and details widget
* Improvement of custom widget actions
* Little upgrade of widget ui

### CODE REVIEW

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseProgressAction.java [hunk 1]
```diff
@@ -42,25 +42,26 @@
...
     public String view() {
         try {
-
             String frontEndMilestonesDataIn = extractWidgetConfig("frontEndMilestonesData");
             this.setFrontEndMilestonesData(frontEndMilestonesDataIn);
             String channelIn = extractWidgetConfig("channel");
             this.setChannel(channelIn);
-
             KieBpmConfig config = formManager.getKieServerConfigurations().get(this.getKnowledgeSourceId());
+            if (null == config) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseProgressAction.java]: Needs test

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceSelectorAction.java [hunk 2]
```diff
@@ -47,24 +48,30 @@ public String selectCaseInstance() {
...
             this.setChannel(channelIn);
-
             this.setKnowledgeSourceId(frontEndCaseDataInjs.getString("knowledge-source-id"));
             this.setContainerid(frontEndCaseDataInjs.getString("container-id"));
             this.setChannelPath(this.getChannel());
-
-            KieBpmConfig config = formManager.getKieServerConfigurations().get(this.getKnowledgeSourceId());
-            this.setCases(caseManager.getCaseInstancesList(config, this.getContainerid()));
-
+            KieBpmConfig config = this.formManager.getKieServerConfigurations().get(this.getKnowledgeSourceId());
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceSelectorAction.java]: Needs test

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceRolesAction.java [hunk 3]
```diff
@@ -47,15 +47,17 @@ public String view() {
             this.setFrontEndCaseData(frontEndCaseDataIn);
             String channelIn = extractWidgetConfig("channel");
             this.setChannel(channelIn);
-
             KieBpmConfig config = formManager.getKieServerConfigurations().get(this.getKnowledgeSourceId());
+            if (null == config) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceRolesAction.java]: Needs test

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceFileAction.java [hunk 4]
```diff
@@ -45,15 +45,17 @@ public String view() {
             this.setFrontEndCaseData(frontEndCaseDataIn);
             String channelIn = extractWidgetConfig("channel");
             this.setChannel(channelIn);
-
             KieBpmConfig config = formManager.getKieServerConfigurations().get(this.getKnowledgeSourceId());
+            if (null == config) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceFileAction.java]: Needs test

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceCommentsAction.java [hunk 5]
```diff
@@ -41,15 +41,17 @@ public String view() {
             this.setFrontEndCaseData(frontEndCaseDataIn);
             String channelIn = extractWidgetConfig("channel");
             this.setChannel(channelIn);
-
             KieBpmConfig config = formManager.getKieServerConfigurations().get(this.getKnowledgeSourceId());
+            if (null == config) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceCommentsAction.java]: Should this be in the base action? Seems repeated across all of the widgets. Maybe a call to the super class for the config to manage this error condition so we don't have the same block in every widget?

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceChartAction.java [hunk 6]
```diff
@@ -39,15 +39,17 @@ public String view() {
             this.setFrontEndCaseData(frontEndCaseDataIn);
             String channelIn = extractWidgetConfig("channel");
             this.setChannel(channelIn);
-
             KieBpmConfig config = formManager.getKieServerConfigurations().get(this.getKnowledgeSourceId());
+            if (null == config) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceChartAction.java]: Move to BaseAction

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by joewhite101** [comment]
Retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by alepintus** [comment]
Retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by alepintus** [comment]
Retest this please

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
All is Well

**Comment by alepintus** [comment]
Retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is Well

**Pull request merged**